### ISSUE TITLE
Implement the design for the responsive menu with centered logo

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#441]
With the new design of the header, the responsive design also needs to be updated/implemented.

Figma link for the header with and without logo:
https://www.figma.com/file/OEtsCj8oD8JPVx4OvsS0U2/Twenty-Twenty-One?node-id=487%3A17

**Site title:**  Font size: 16px, font weight: 500, line-height: 115%;
**Tagline:** Font size: 16px, font-weight; normal (400), line-height: 160%;

The right edge of the button **icon** aligns with the right edge of the border below the logo.

The focus order needs to match the visual order, depending on if the logo or title are linked and enabled:
Linked logo
Linked site title
Menu toggle

**Comment by aristath** [comment]
Thank you @carolinan :+1: 

Posting screenshots 'cause I can't stand working with Figma and it makes my laptop wanna take off :sweat_smile: 

![Screenshot_2020-10-13 Figma(2)](https://user-images.githubusercontent.com/588688/95828388-9b069c80-0d3d-11eb-995a-9864a660b0d2.png)
![Screenshot_2020-10-13 Figma(1)](https://user-images.githubusercontent.com/588688/95828405-9e9a2380-0d3d-11eb-92b8-0d0c7ad4a5df.png)
![Screenshot_2020-10-13 Figma](https://user-images.githubusercontent.com/588688/95828409-9f32ba00-0d3d-11eb-9c7d-9e5b455fc713.png)

**Comment by carolinan** [comment]
I don't believe the first, top one is meant to be used, unless there is meant to be an **option** for the position of the logo?
How would we know when to left align or center the logo?

**Comment by aristath** [comment]
>  How would we know when to left align or center the logo?

If on mobile, it would be really weird to have the logo centered and the menu on the right... I think it would be reasonable to assume that if there is a logo and no title/tagline, the logo should be on the opposite side of the menu. It is more visually balanced than having the logo on the center and the menu on the right...

**Comment by ryelle** [comment]
> How would we know when to left align or center the logo?

It depends on whether there's a site title or not, which is how it works on trunk right now.

> I think it would be reasonable to assume that if there is a logo and no title/tagline, the logo should be on the opposite side of the menu.

Exactly :) 

So the main issue with the mobile layout is that the menu button isn't in the right place?

![Screen Shot 2020-10-13 at 10 32 34](https://user-images.githubusercontent.com/541093/95875003-8925f900-0d3f-11eb-8302-c7f461a68813.png)

**Comment by aristath** [comment]
@ryelle I have a PR on #442 which hopefully addresses the issue and implements the layouts from above, it just needs a thorough review :heart:

**Comment by ryelle** [comment]
Looking at it now 👍   I just wasn't sure how different this was from what was implemented in #435.
